### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by ysimillides**
added more parameters to the assemble, added a few linalg commands

### CODE REVIEW

File: src/jfem.jl [hunk 1]
```diff
@@ -61,13 +59,15 @@ export dx, ds,dS,dP
 
 
 @fenicsclass Matrix
-assemble(assembly_item::Form)=Matrix(fenics.assemble(assembly_item.pyobject)) #this gives as PETScMatrix of appopriate dimensions
+#assemble(assembly_item::Form)=Matrix(fenics.assemble(assembly_item.pyobject))
+assemble(assembly_item::Form;tensor=nothing, form_compiler_parameters=nothing, add_values=false, finalize_tensor=true, keep_diagonal=false, backend=nothing) = Matrix(
+fenics.assemble(assembly_item.pyobject,tensor=tensor,form_compiler_parameters=form_compiler_parameters,add_values=add_values,finalize_tensor=finalize_tensor,keep_diagonal=keep_diagonal,backend=backend))#this gives as PETScMatrix of appopriate dimensions
```
Comment (ChrisRackauckas) [inline src/jfem.jl]: is a PETScMatrix a type we have? Should we depend on PETSc.jl and wrap this somehow?

**Comment by ysimillides** [comment]
I think it comes with fenics  ( but i can also check out petsc.jl)
________________________________

**Comment by ChrisRackauckas** [comment]
Looks good, though how did the filename come along? Did that come from FEniCS? It seems like it should be named linalg.jl or assembly.jl or something like that.

**Comment by ysimillides** [comment]
The name came from https://fenicsproject.org/olddocs/dolfin/2016.2.0/python/programmers-reference/fem/index.html .
assembly.jl might also be a good name (whereas linalg.jl might indicate it is purely for linear algebra which isn't  really the case imo)

**Comment by ChrisRackauckas** [comment]
Sounds good for now since it's mirroring dolfin. It could always be changed later.

**Pull request merged**